pal/zephyros/pal_i2c.c

```diff
@@ -0,0 +1,338 @@
+
+pal_status_t pal_i2c_init(const pal_i2c_t * p_i2c_context)
+{
+    pal_zephyr_i2c_t *p_i2c_ctx = (pal_zephyr_i2c_t*)(p_i2c_context->p_i2c_hw_config);
+    p_i2c_ctx->p_i2c_dev = device_get_binding(I2C_PORT_NAME);
+    
+    if (p_i2c_ctx->p_i2c_dev == 0)
+    {
+        LOG_ERR("could not bind I2C-device");
+        return PAL_STATUS_FAILURE;
+    }
+
+    return PAL_STATUS_SUCCESS;
+}
+
+pal_status_t pal_i2c_deinit(const pal_i2c_t * p_i2c_context)
+{
+    return PAL_STATUS_SUCCESS;
+}
+
+pal_status_t pal_i2c_write(pal_i2c_t * p_i2c_context, uint8_t * p_data, uint16_t length)
+{
+    int lock_result;
+    int i2c_write_result;
+    pal_status_t status = PAL_STATUS_FAILURE;
+
+    pal_zephyr_i2c_t *p_i2c_ctx = (pal_zephyr_i2c_t*)(p_i2c_context->p_i2c_hw_config);
+
+	if((p_i2c_context != NULL) && (p_i2c_ctx->p_i2c_dev == NULL))
+	{
+		pal_i2c_init(p_i2c_context);
+	}
+
+    //Acquire the I2C bus before read/write
+    lock_result = k_sem_take(&i2c_bus_lock, K_NO_WAIT);
+
+    if (lock_result == BUS_ENTRY_SUCCESS)
+    {
+        gp_pal_i2c_current_ctx = p_i2c_context;
+
+        //Invoke the low level i2c master driver API to write to the bus
+
+        i2c_write_result = i2c_write(p_i2c_ctx->p_i2c_dev, p_data, (u32_t)length, p_i2c_context->slave_address);
+        //printk("[PAL]: I2C write status %d\n", i2c_write_result);
+        
+        if(i2c_write_result != 0)
+        {
+            //If I2C Master fails to invoke the write operation, invoke upper layer event handler with error.
+            
+            //lint --e{611} suppress "void* function pointer is type casted to upper_layer_callback_t type"
+
+            ((upper_layer_callback_t)(p_i2c_context->upper_layer_event_handler))
+                                                       (p_i2c_context->p_upper_layer_ctx , PAL_I2C_EVENT_ERROR);
+        }
+        else
+        {
+			// !!!OPTIGA_LIB_PORTING_REQUIRED
+			/**
+			* Infineon I2C Protocol is a polling based protocol, if foo_i2c_write will fail it will be reported to the 
+			* upper layers by calling 
+			* (p_i2c_context->upper_layer_event_handler))(p_i2c_context->p_upper_layer_ctx , PAL_I2C_EVENT_ERROR);
+			* If the function foo_i2c_write() will succedd then two options are possible
+			* 1. if foo_i2c_write() is interrupt based, then you need to configure interrupts in the function 
+			*    pal_i2c_init() so that on a succesfull transmit interrupt the callback i2c_master_end_of_transmit_callback(),
+			*    in case of successfull receive i2c_master_end_of_receive_callback() callback 
+			*    in case of not acknowedged, arbitration lost, generic error i2c_master_nack_received_callback() or
+			*    i2c_master_arbitration_lost_callback()
+			* 2. If foo_i2c_write() is a blocking function which will return either ok or failure after transmitting data
+			*    you can handle this case directly here and call 
+			*    invoke_upper_layer_callback(gp_pal_i2c_current_ctx, PAL_I2C_EVENT_SUCCESS);
+			*    
+			*/

```

Comment:
I suggest to remove this

Reply:
Yes, sure, I forgot about these comments. I will go through the whole code and remove them where I think they are no longer needed.